src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs

```diff
@@ -74,18 +66,22 @@ public Result OnStartup(UIControlledApplication application)
 
             Products = new List<DynamoProduct>();
             int index = -1;
+            var selectedVersion = selectorData.SelectedVersion.ToString(2);
             foreach (var p in dynamoProducts)
             {
                 var path = VersionLoader.GetDynamoRevitPath(p, revitVersion);
                 if (!File.Exists(path))
                     continue;
 
-                if (path.Equals(cachedPath))
+                if (p.VersionInfo.ToString(2) == selectedVersion)
```

Comment:
It is better to make it consistent to take 3 fields like what is done in the AddSplitButtonGroup function. Or is this done for a reason?